lib/Crypto/DecryptAll.php

```diff
@@ -90,55 +90,77 @@ public function __construct(
+
+			// Check a method has been passed
+			if(!$input->hasOption('method') && !in_array($input->getOption('method'),['recovery', 'password'])) {
+				$output->writeln('A method must be supplied when decrypting from user-key state');
+				return false;
+			}
+			if (empty($user)) {
+				// all users, so only recovery is possible
+				if($input->getOption('method')==='recovery' && !empty(getenv('OC_RECOVERY_PASSWORD'))) {
+					// Then we can attempt to use this for all of the users
+					$output->writeln('Attempting to use recovery key from environment for all users. Users must have enabled recovery keys for this to work.');
+					$password = \getenv('OC_RECOVERY_PASSWORD');
 				} else {
-					$user = $recoveryKeyId;
+					$output->writeLn('Recovery key is the only supported method for decrypting all users in one command. The key is read from OC_RECOVERY_PASSWORD.');
+					return false;
 				}
 			} else {
-				$output->writeln('You can only decrypt the files of all users if the');
-				$output->writeln('recovery key is enabled by the admin and activated by the users.');
-				$output->writeln('');
-				$user = $recoveryKeyId;
-			}
-
-			$question->setHidden(true);
-			$question->setHiddenFallback(false);
-			$password = $this->questionHelper->ask($input, $output, $question);
+				// Specific user, password is an option here
+				if($input->getOption('method')==='recovery' && !empty(getenv('OC_RECOVERY_PASSWORD'))) {
+					if ($this->util->isRecoveryEnabledForUser($user) === false) {
+						$output->writeln('Password recovery is not enabled for ' . $user);
+						return false;
+					}
+					// Then we can attempt to use this for all of the users
+					$output->writeln('Attempting to use recovery key from environment: OC_RECOVERY_PASSWORD');
+					$password = getenv('OC_RECOVERY_PASSWORD');
+
+					try {
+						if ($this->keyManager->checkRecoveryPassword($password) === false) {
+							$throwLoginException = true;
+						}
+					} catch (\Exception $e) {
+						$throwLoginException = true;
+					}
 
+					$recoveryKeyId = $this->keyManager->getRecoveryKeyId();
+					$user = $recoveryKeyId;
+				} elseif ($input->getOption('method')==='password' && !empty(getenv('OC_PASSWORD'))) {
+					$output->writeln('Attempting to use users password from environment: OC_PASSWORD');
+					// Then we want to use the users password and it has been supplied
+					$password = getenv('OC_PASSWORD');
 
-			$throwLoginException = false;
-			if ($recoveryKeyId === $user) {
-				try {
-					if ($this->keyManager->checkRecoveryPassword($password) === false) {
+					if ($this->userManager->checkPassword($user, $password) === false) {
 						$throwLoginException = true;
 					}
-				} catch (\Exception $e) {
-					$throwLoginException = true;
+				} elseif ($input->getOption('method') === null) {
+					//Grab user specific password
+					$question = new Question('Please enter the user: ' . $user . ' login password: ');
```

Comment:
"Please enter the login password from user $user: "

Reply:
Done.